aiida_mlip/parsers/train_parser.py

```diff
@@ -0,0 +1,117 @@
+"""
+Parser for mlip train.
+"""
+
+from pathlib import Path
+
+from ase.io import read
+
+from aiida.engine import ExitCode
+from aiida.orm import Dict, FolderData
+from aiida.orm.nodes.process.process import ProcessNode
+from aiida.parsers.parser import Parser
+
+from aiida_mlip.data.model import ModelData
+from aiida_mlip.helpers.converters import convert_numpy
+
+
+class TrainParser(Parser):
+    """
+    Parser class for parsing output of calculation.
+
+    Parameters
+    ----------
+    node : aiida.orm.nodes.process.process.ProcessNode
+        ProcessNode of calculation.
+
+    Methods
+    -------
+    __init__(node: aiida.orm.nodes.process.process.ProcessNode)
+        Initialize the SPParser instance.
+
+    parse(**kwargs: Any) -> int:
+        Parse outputs, store results in the database.
+
+    Returns
+    -------
+    int
+        An exit code.
+
+    Raises
+    ------
+    exceptions.ParsingError
+        If the ProcessNode being passed was not produced by a singlePointCalculation.
+    """
+
+    def __init__(self, node: ProcessNode):
+        """
+        Check that the ProcessNode being passed was produced by a `Singlepoint`.
+
+        Parameters
+        ----------
+        node : aiida.orm.nodes.process.process.ProcessNode
+            ProcessNode of calculation.
+        """
+        super().__init__(node)
+
+    # disable for now
+    # pylint: disable=too-many-locals
+    def parse(self, **kwargs) -> int:
+        """
+        Parse outputs, store results in the database.
+
+        Parameters
+        ----------
+        **kwargs : Any
+            Any keyword arguments.
+
+        Returns
+        -------
+        int
+            An exit code.
+        """
+        remote_dir = Path(self.node.get_remote_workdir())
+        mlip_dict = self.node.get_option("mlip_config").as_dictionary()
+        log_dir = remote_dir / Path(mlip_dict.get("log_dir", "logs"))
+        checkpoint_dir = remote_dir / Path(
+            mlip_dict.get("checkpoint_dir", "checkpoints")
+        )
+        results_dir = remote_dir / Path(mlip_dict.get("results_dir", "results"))
+        model_dir = remote_dir / Path(mlip_dict.get("model_dir", ""))
```

Comment:
Looking at this, might it be better to have:
```suggestion
        remote_dirs = {typ: remote_dir / mlip_dict.get(f"{typ}_dir", default)
                       for typ, default in (("log", "logs"), 
                                            ("checkpoint", "checkpoints"),
                                            ("results", "results"),
                                            ("model", ""))}
```
And use the dict as the main reference for these.